### PR TITLE
PLAT-105929/Flow-Service-API-Enhancements [READY]

### PR DESCRIPTION
**Pull request by carlojavier**
Overall improvements to the Flow Service API reference

## Description

Overall improvements to the Flow Service API reference

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/swagger-specs/flow-service.yaml [hunk 1]
```diff
@@ -243,20 +239,20 @@ paths:
...
           name: body
           description: >-
-            - **op** - **(Required)** The operation call used to define the action needed to update the connection. Operations include `add`, `replace`, and `remove`.
-
-            - **path** - (Optional) The path of the parameter to be updated.
-
-            - **value** - **(Required)** The new value you want to update your parameter with.
+            **op** - **(Required)** The operation call used to define the action needed to update the connection. Operations include `add`, `replace`, and `remove`.
+            
+            **path** - (Optional) The path of the parameter to be updated.
```
Comment (vivetiwa) [inline src/swagger-specs/flow-service.yaml:253]: `path` is required parameter for update operation.

File: src/swagger-specs/flow-service.yaml [hunk 2]
```diff
@@ -35,26 +35,27 @@ host: platform.adobe.io
 basePath: /data/foundation/flowservice
 tags:
   - name: Connections
-    description: Connections retain information regarding how to connect to a source or target. This includes a unique connection ID, which is necessary in order to explore or navigate files and objects.
+    description: Base connections retain information between your source and Adobe Experience Platform. They include metadata, a partial view of the credentials used to authenticate a given source, and the unique base connection ID that corresponds to the queried connection. The base connection ID represents your authenticated connection and allows you to explore and inspect the type, structure, and contents of the data that you want to bring to Experience Platform. The base connection ID is required to create a source connection. An instance of a base connection is specific to an organization.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:38]: Would this still mean the same if I changed the order of the two sentences?

"The base connection ID is required to create a source connection. It also represents your authenticated connection and allows you to explore and inspect the type, structure, and contents of the data that you want to bring to Experience Platform. "

File: src/swagger-specs/flow-service.yaml [hunk 3]
```diff
@@ -35,26 +35,27 @@ host: platform.adobe.io
...
+    description: Connection specifications retain the connector properties of a given source. A connection spec can be divided into three distinct sections, authentication specifications (authSpec), source specifications (sourceSpec), and explore specifications (exploreSpec). The authSpec defines the authentication type(s) that you can use to connect your source to Experience Platform. The sourceSpec defines how you can interact with your source, including pagination types, scheduling formats, and query parameters that a given source supports. The sourceSpec also includes the schema that defines the attributes of a source. The exploreSpec defines the structure of returned data when a source is queried against. Connection specs include a connection spec ID. This ID is fixed and each source on Experience Platform has its own fixed connection spec ID. This ID is required to create a base connection and a source connection. Connection specs are organization agnostic.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:40]: Probably needs a colon. "...into three distinct sections: authentication specifications..."

File: src/swagger-specs/flow-service.yaml [hunk 4]
```diff
@@ -35,26 +35,27 @@ host: platform.adobe.io
...
   - name: Source connections
-    description: Source connections create and manage a connection to the external source from where data is ingested. Source connections contain information like data source, data format, and the source connection ID needed to create a flow. A source connection instance is specific to a tenant and IMS Organization.
+    description: Source connections represent the incoming data from a given source. Source connections define where the data is coming from and in what format. Source connections also refer to the base connection for authentication information. For file-based sources like cloud storages, source connections can define additional information on the data transfer, such as column delimiter, encoding type, compression type, regular expressions, and whether ingestion is to be done recursively. An instance of a source connection is specific to an organization.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:48]: Can we rephrase it to: "Source connections define the origin and format of the data."

File: src/swagger-specs/flow-service.yaml [hunk 5]
```diff
@@ -71,17 +72,18 @@ paths:
         - $ref: '#/parameters/count'
       responses:
         '200':
-          description: Returns a list of connections in JSON format.
+          description: Successful operation. A successful response returns a list of base connections that match the request parameters.
           schema:
             $ref: '#/definitions/ConnectionListResponse'
         '401':
-          description: OAuth token is missing.
+          description: The The OAuth token is missing.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:79]: There's an extra "The."

File: src/swagger-specs/flow-service.yaml [hunk 6]
```diff
@@ -142,26 +145,27 @@ paths:
...
           in: path
           description:
-            "The ID of the connection you wish to retrieve."
+            "The ID of the base connection you want to retrieve."
           required: true
           type: string
       responses:
         '200':
-          description: Returns details for specified base connection in JSON format
+          description: Returns details for the queried base connection in JSON format
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:156]: The sentence is missing a period.

File: src/swagger-specs/flow-service.yaml [hunk 7]
```diff
@@ -203,35 +207,27 @@ paths:
...
-          description: Returns file structure of specified connection ID, including details regarding type, name, path and if file can be previewed or fetched.
-          examples:
-            application/json:
-              items:
-                - type: "folder"
-                  name: "test-connectors"
-                  path: "test-connectors"
-                  canPreview: "false"
-                  canFetchSchema: "false"
+          description: Returns file structure of a specified base connection ID, including details regarding type, name, path and whether the file can be previewed or fetched.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:218]: "Returns _the_ file structure..."

File: src/swagger-specs/flow-service.yaml [hunk 8]
```diff
@@ -203,35 +207,27 @@ paths:
...
         '503':
           description: The service is unavailable.
-  '/connections/{CONNECTION_ID}/':
+  '/connections/{BASE_CONNECTION_ID}/':
     patch:
       tags:
         - Connections
       summary: Update an existing connection
-      description: You can use this endpoint to update the information about an existing connection.
+      description: Update the information of your base connection by making PATCH request to the `/connections/{BASE_CONNECTION_ID}` endpoint while providing the ID of the base connection that you want to update.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:230]: "...by making _a_ PATCH request..."

File: src/swagger-specs/flow-service.yaml [hunk 9]
```diff
@@ -379,13 +375,13 @@ paths:
         - $ref: '#/parameters/count'
       responses:
         '200':
-          description: Returns a list of flows in JSON format.
+          description: Returns a list of flows that match the request parameteres in JSON format.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:378]: Parameters spelling has a typo

File: src/swagger-specs/flow-service.yaml [hunk 10]
```diff
@@ -1032,88 +975,88 @@ definitions:
     properties:
       specName:
         type: string
-        example: "Basic Authentication for a cloud storage source"
+        example: "Basic Authentication for google-cloud"
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:978]: Should Google be capitalized?

File: src/swagger-specs/flow-service.yaml [hunk 11]
```diff
@@ -1311,172 +1256,378 @@ definitions:
...
+        description: The name of the type of the source.
       type:
         type: string
-        example: "object"
-      description:
-        type: string
-        example: "Defines params required when this connection is used as a source in flow"
-      path:
+        example: "CloudStorage"
+        description: The type of the source. This value is the same as the category assigned to the source.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:1477]: "The type of source." I'm not 100% sure if "the" should precede "source" or not.

File: src/swagger-specs/flow-service.yaml [hunk 12]
```diff
@@ -1311,172 +1256,378 @@ definitions:
...
-          default:
-            type: string
-            example: "false"
+            example: "object"
           description:
             type: string
-            example: "Indicates recursive copy in case of folder or wild card path, default is false"
-      policy:
-        $ref: '#/definitions/PolicyItemDTO'
+            example: "Defines params required when this connection is used as a source in flow"
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:1490]: The sentence is missing a period.

File: src/swagger-specs/flow-service.yaml [hunk 13]
```diff
@@ -1311,172 +1256,378 @@ definitions:
...
+        description: The organization ID to which the source connection belongs to.
       name:
         type: string
-        example: "IT-S3 Flow"
+        example: "Google Cloud Storage Flow"
         description: The name of the flow.
       description:
         type: string
-        example: "Flow created during integration tests"
+        example: "A Google Cloud Storage flow for ACME customer data"
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:1602]: Some examples are missing periods.

File: src/swagger-specs/flow-service.yaml [hunk 14]
```diff
@@ -1677,40 +2031,311 @@ definitions:
...
+            properties:
+              schema:
+                type: string
+                example: "http://json-schema.org/draft-07/schema#"
+              type:
+                type: string
+                example: "object"
+              description:
+                type: string
+                example: "defines various params required for different mapping from source to target"
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:2114]: Define should probably be capitalized and the sentence is missing a period.

File: src/swagger-specs/flow-service.yaml [hunk 15]
```diff
@@ -2126,55 +2768,43 @@ parameters:
...
+    description: Sort the listed resources by specified fields. For example, `orderby=name` sorts results by name in ascending order (A-Z). Adding a dash (`-`) before the field name (`orderby=-name`) sorts the results in descending order (Z-A).
     required: false
     type: string
-  orderby:
-    name: orderby
+  limit:  
+    name: limit
     in: query
-    description: "Sort response by specified fields separated by `,` and/or prefix field with `-` for descending order."
+    description: Restrict the maximim number of items returned for a given request. For example, adding a parameter of `limit=10` in a request to retrieve all base connections returns only a maximum of 10 base connections. When used to query the API, the returned response automatically includes a `continuationToken` that can be used to fetch records that were not returned due to the limits applied by the parameter.
```
Comment (serahmetpalli) [inline src/swagger-specs/flow-service.yaml:2789]: Typo - "...maximum."

File: src/swagger-specs/flow-service.yaml (hunk 10 above)
Comment (carlojavier) [reply src/swagger-specs/flow-service.yaml:978]: in theory yes, but in this case, "Basic Authentication for google-cloud" is the actual coded name of the specName so not this one

**Review by vivetiwa — APPROVED** [review]
LGTM. thanks!
